state/upgrades_test.go

```diff
@@ -3212,6 +3212,77 @@ func (s *upgradesSuite) TestUpdateK8sModelNameIndex(c *gc.C) {
 	)
 }
 
+func (s *upgradesSuite) TestAddControllerNodeDocs(c *gc.C) {
+	machinesColl, closer := s.state.db().GetRawCollection(machinesC)
+	defer closer()
+	controllersColl, closer2 := s.state.db().GetRawCollection(controllersC)
+	defer closer2()
+
+	// Will will never have different UUIDs in practice but testing
+	// with that scenario avoids any potential bad cod assumptions.
+	uuid1 := "uuid1"
+	uuid2 := "uuid2"
+	err := machinesColl.Insert(bson.M{
+		"_id":       ensureModelUUID(uuid1, "1"),
+		"machineid": "1",
+		"novote":    false,
+		"hasvote":   true,
+	}, bson.M{
+		"_id":       ensureModelUUID(uuid1, "2"),
+		"machineid": "2",
+		"novote":    false,
+		"hasvote":   false,
+	}, bson.M{
+		"_id":       ensureModelUUID(uuid1, "3"),
+		"machineid": "3",
+		"novote":    true,
+		"hasvote":   false,
+	}, bson.M{
+		"_id":       ensureModelUUID(uuid2, "1"),
+		"machineid": "1",
+		"novote":    false,
+		"hasvote":   false,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	controllerSettings := bson.M{}
+	for k, v := range coretesting.FakeControllerConfig() {
+		controllerSettings[k] = v
+	}
+	expected := bsonMById{
+		{
+			"_id": "apiHostPorts", "apihostports": []interface{}{},
```

Comment:
Wow, I didn't realise how much random stuff there was in the controllers collection! Wouldn't it be easier to use a different collection for controller nodes, rather than jamming them into controllersC?

Reply:
ok